File: contest-management-client-server-java/ContestMgmtPersistence/src/main/java/contestmgmt/persistence/repository/IParticipantRepository.java
package contestmgmt.persistence.repository;

import contestmgmt.model.Competition;
import contestmgmt.model.Participant;

public interface IParticipantRepository extends ICrudRepository<Long, Participant> {
    Iterable<Participant> findParticipantsByCompetition(Competition competition);

    Iterable<Long> findParticipantIdsByName(String firstName, String lastName);
}

File: contest-management-client-server-cs/persistence/IParticipantRepository.cs
using System.Collections.Generic;
using model;

namespace persistence
{
    public interface IParticipantRepository : ICrudRepository<long, Participant>
    {
        IEnumerable<Participant> FindParticipantsByCompetition(long competitionId);
        
        IEnumerable<long> FindParticipantIdsByName(string firstName, string lastName);
    }
}